python/Ganga/Core/GangaRepository/GangaRepositoryXML.py

```diff
@@ -239,11 +239,12 @@ def shutdown(self):
         from Ganga.Utility.logging import getLogger
         logger = getLogger()
         logger.debug("Shutting Down GangaRepositoryLocal: %s" % self.registry.name)
-        for k in self._fully_loaded:
-            try:
-                self.index_write(k, shutdown=True)
-            except Exception as err:
-                logger.error("Warning: problem writing index object with id %s" % k)
+        for k in self.objects:
+            if self.objects[k]._inMemory:
```

Comment:
@KonstantinSchubert I would say the main places where we're focusing development ahead of 6.2.0 is fixing bugs/missing features in `GaudiExec` which has now superceded `GaudiRun` as well as fixing any minor issues which get in the way of a user doing something.
I imagine 6.2.0 will be about 6.1.25 or so. 
(Gauss+GaussSplitter are the main missing features that come to mind in addition to some discussion which needs to happen first around Bender/OStap support)

Shortly after 6.2.0 I think we plan to integrate the new credentials branch which will address several issues behind the scenes with regard to handling proxies and such in a better way.
Then once all of this is out of the way I was hoping to see some code rewritten around the loading/lazy-loading of objects from disk in addition to the new objects to replace the `Executable` application which exposes problems.
In addition to this the IGangaFiles need to support workerDir, localDir properly and be made to behave the same way over the next few releases which will make handling of files much easier in the long term.

(This isn't a completely exhaustive list but is a run down of some of the key development tasks that I'm aware of over the next few months).